src/main/scala/org/constellation/infrastructure/endpoints/SnapshotEndpoints.scala

```diff
@@ -166,6 +171,37 @@ class SnapshotEndpoints[F[_]](implicit F: Concurrent[F]) extends Http4sDsl[F] {
         .flatMap(Ok(_))
   }
 
+  private def postSnapshotProposal(
+    snapshotProposalGossipService: SnapshotProposalGossipService[F],
+    redownloadService: RedownloadService[F]
+  ): HttpRoutes[F] =
+    HttpRoutes.of[F] {
+      case req @ POST -> Root / "peer" / "snapshot" / "created" =>
+        for {
+          message <- req.as[GossipMessage[SnapshotProposalGossip]]
+          data = message.data
+          senderId = req.headers.get(`X-Id`).map(_.value).map(Id(_)).get
+
+          res <- snapshotProposalGossipService.validate(message, senderId) match {
+            case Left(EndOfCycle)                => snapshotProposalGossipService.finishCycle(message) >> Ok()
+            case Left(IncorrectReceiverId(_, _)) => BadRequest()
+            case Left(IncorrectSenderId(_))      => Response[F](status = Unauthorized).pure[F]
+            case Left(_)                         => InternalServerError()
```

Comment:
I think we can log an error here, as 500-response does not return it. wdyt?

Reply:
Or we should even log all the Left errors 🤔